packages/rpc/README.md

```diff
@@ -14,4 +14,108 @@
+
+This is useful in cases where you need to make assertions about the suitablilty of a RPC for a given purpose. For example, you might like to make it a type error to combine certain types with RPC belonging to certain clusters, at compile time.
+
+```ts
+async function getSpecialAccountInfo(
+    address: Address<'ReAL1111111111111111111111111111'>,
+    rpc: RpcMainnet,
+): Promise<SpecialAccountInfo>;
+async function getSpecialAccountInfo(
+    address: Address<'TeST1111111111111111111111111111'>,
+    rpc: RpcDevnet | RpcTestnet,
+): Promise<SpecialAccountInfo>;
+async function getSpecialAccountInfo(address: Address, rpc: Rpc): Promise<SpecialAccountInfo> {
+    /* ... */
+}
+const rpc = getSolanaRpc({ url: devnet('https://api.devnet.solana.com') });
+await getSpecialAccountInfo(address('TeST1111111111111111111111111111'), rpc); // ERROR
+```
+
+### `RpcFromTransport<TRpcMethods, TRpcTransport extends RpcTransport>`
+
+Given a `RpcTransport`, this utility type will resolve to as specific a `Rpc` as possible.
+
+```ts
+function createCustomRpc<TRpcTransport extends RpcTransport>(
+    transport: TRpcTransport,
+): RpcFromTransport<MyCustomRpcMethods, TRpcTransport> {
+    /* ... */
+}
+const transport = createDefaultRpcTransport({ url: mainnet('http://rpc.company') });
+transport satisfies RpcTransportMainnet; // OK
+const rpc = createCustomRpc(transport);
+rpc satisfies RpcMainnet<MyCustomRpcMethods>; // OK
+```
+
+### SolanaRpcApiFromTransport<TTransport extends RpcTransport>
+
+## Constants
+
+### `DEFAULT_RPC_CONFIG`
+
+When you create `Rpc` instances with custom transports but otherwise the default RPC API behaviours, use this.
+
+```ts
+const myCustomRpc = createRpc({
+    api: createSolanaRpcApi(DEFAULT_RPC_CONFIG),
+    transport: myCustomTransport,
+});
+```
+
+## Functions
+
+### `createDefaultRpcTransport(config)`
+
+Creates a `RpcTransport` with some default behaviours.
+
+The default behaviours include:
+
+-   An automatically-set `Solana-Client` request header, containing the version of `@solana/web3.js`
+-   Logic that coalesces multiple calls in the same runloop, for the same methods with the same arguments, into a single network request.
+
+#### Arguments
+
+A config object with the following properties:
+
+-   `dispatcher_NODE_ONLY`: An optional `Undici.Dispatcher` instance that governs how the networking stack should behave. This option is only relevant in Node applications. Consult the documentation for the various subclasses of `Undici.Dispatcher`, such as `Agent`, `Client`, and `Pool`, at https://undici.nodejs.org/#/docs/api/Client.
+-   `headers`: An optional object where the keys are HTTP header names and the values are HTTP header values. This parameter is typed to disallow certain headers from being overwritten.
+-   `url`: A `ClusterUrl` at which the RPC server can be contacted.
+
+### `createSolanaRpc(clusterUrl, config)`
```

Comment:
This is possibly the most important function of this whole package and perhaps we should find a way to bump its visibility on the readme. I know we're following a "Types" then "Functions" documentation pattern but maybe we can make an exemption here? Or just mention it in the intro with a link to it further down the page.